### ISSUE TITLE
about diff

### ISSUE THREAD

**Issue by zjz5250** [xiaolonw/caffe-video_triplet#11]
@xiaolonw        
                if(tloss1 > 0)
			{
				for(int k = 0; k < dim; k ++)
			    {
					fori_diff[k] += (fneg[k] - fpos[k]); // / (pairNum * 1.0 - 2.0);
					fpos_diff[k] += -fori[k]; // / (pairNum * 1.0 - 2.0);
					fneg_diff[k] +=  fori[k];
			    }
			}
			if(tloss2 > 0)
			{
				for(int k = 0; k < dim; k ++)
				{
					fori_diff[k] += -fpos[k]; // / (pairNum * 1.0 - 2.0);
				    fpos_diff[k] += fneg[k]-fori[k]; // / (pairNum * 1.0 - 2.0);
				    fneg_diff[k] += fpos[k];
				}
			}
 为什么loss1与loss2大于0时，计算梯度的方式不一致呢？
loss = f(a).f(n)-f(a).f(p)
那求偏导后，loss1>0是，diff的计算与上述loss相同，loss2>0时的计算公式是怎么得来呢？
多谢啦
